prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/hx_insight/PredictionCollector.java

```diff
@@ -97,64 +86,65 @@ public void configure()
+            .loopDoWhile(statusCodeNot204())
                 .toD(batchesUrl)
-                .log(DEBUG, log, "Processing prediction batches: ${body}")
-                .choice().when(bodyNotEmpty())
+                .choice().when(statusCodeNot204())
+                    .log(DEBUG, log, "Processing prediction batches page ${headers.%s} started".formatted(BATCHES_PAGE_NO_HEADER))
                     .unmarshal(predictionsBatchDataFormat)
                     .split(body())
-                        .setHeader(BATCH_ID_HEADER, simple("${body.id}"))
-                        // notify HxI -> started batch processing
-                        .setHeader(PREDICTIONS_PAGE_NO_HEADER, constant(1))
-                        .loopDoWhile(bodyNotEmpty())
-                            .log(DEBUG, log, "Calling URL: " + predictionsUrl)
-                            .toD(predictionsUrl)
-                            .log(DEBUG, log, "Sending predictions to internal buffer: ${body}")
-                            .choice().when(bodyNotEmpty())
-                                .unmarshal(predictionsDataFormat)
-                                .split(body())
-                                    .marshal(predictionDataFormat)
-                                    .log(TRACE, log, "Sending prediction to internal buffer: ${body}")
-                                    .to(insightPredictionsProperties.bufferEndpoint())
-                                .end()
-                            .end()
-                            .setHeader(PREDICTIONS_PAGE_NO_HEADER).spel("#{request.headers['%s'] + 1}".formatted(PREDICTIONS_PAGE_NO_HEADER))
-                        .end()
-                        // notify HxI -> finished batch processing
+                        .to(BATCH_PROCESSOR_ENDPOINT)
+                    .end()
+                    .setHeader(BATCHES_PAGE_NO_HEADER, spel("#{request.headers['%s'] + 1}".formatted(BATCHES_PAGE_NO_HEADER)))
+                    .toD(batchesUrl)
+                .end()
+            .end()
+            .log(DEBUG, log, "Finished processing predictions");
+
+        from(BATCH_PROCESSOR_ENDPOINT)
+            .routeId(BATCH_PROCESSOR_ROUTE_ID)
+            .log(DEBUG, log, "Processing prediction batch ${body.id} started")
+            .setHeader(BATCH_ID_HEADER, simple("${body.id}"))
+            .setHeader(PREDICTIONS_PAGE_NO_HEADER, constant(1))
+            .loopDoWhile(statusCodeNot204())
+                .toD(predictionsUrl)
+                .choice().when(statusCodeNot204())
+                    .log(TRACE, log, "Processing page ${headers.%s} of predictions in batch ${headers.%s}, ${body}, ${header.CamelHttpResponseCode}".formatted(PREDICTIONS_PAGE_NO_HEADER, BATCH_ID_HEADER))
+                    .unmarshal(predictionsDataFormat)
+                    .split(body())
+                        .marshal(predictionDataFormat)
+                        .to(insightPredictionsProperties.bufferEndpoint())
```

Comment:
Not a change for this PR, but I don't think the queue name should be configurable. We should review all the properties and see what makes sense to ask the user to set.